tests/conftest.py

```diff
@@ -190,6 +178,37 @@ def legacy_opmath_only():
         pytest.skip("This test exclusively tests legacy opmath")
 
 
+#######################################################################
+
+
+@pytest.fixture
+def seed(request):
+    """An integer random number generator seed
+
+    This fixture overrides the ``seed`` fixture provided by pytest-rng, adding the flexibility
+    of locally getting a new seed for a test case by applying the ``local_salt`` marker. This is
+    useful when the seed from pytest-rng happens to be a bad seed that causes your test to fail.
+
+    .. code_block:: python
+
+        @pytest.mark.local_salt(42)
+        def test_something(seed):
+            ...
+
+    The value passed to ``local_salt`` needs to be an integer.
+
+    """
+
+    fixture_manager = request._fixturemanager  # pylint:disable=protected-access
+    fixture_defs = fixture_manager.getfixturedefs("seed", request.node)
+    original_fixture_def = fixture_defs[0]  # the original seed fixture provided by pytest-rng
+    original_seed = original_fixture_def.func(request)
+    marker = request.node.get_closest_marker("local_salt")
+    if marker and marker.args:
+        return original_seed + marker.args[0]
+    return original_seed
```

Comment:
Each time we update the global salt, some tests may fail if it unfortunately receives a bad seed. If we modify the seed for these tests locally in the test function every time this happens, we can imagine that the number of tests with locally modified seeds will increase. The marker is my answer to a more standardized way of getting a new seed in case the original one doesn't work. In the future we can easily do a sweep of the entire test suite to get rid of some temporary local salts wherever possible.

Reply:
@JerryChen97 so `pytest-rng` will distribute a different but consistent seed to each test case, and the `rng_salt` specified in `pytest.ini` is the only control we have over the seeds provided to the test cases. If one test receives an unfortunate bad seed that causes it to fail, we don't want to change the `rng_salt` in `pytest.ini` which will change the seed for every test case in the test suite, we apply a local salt to this test only.